internal/static/integrations/ssh_exporter/ssh_client.go

```diff
@@ -0,0 +1,194 @@
+    }
+    scannedKey := strings.TrimSpace(string(output))
+
+    for _, line := range knownHostsContent {
+        if strings.Contains(line, targetAddress) {
+            if line != scannedKey {
+                return fmt.Errorf(
+                    "host key mismatch for %s: existing key [%s] differs from scanned key [%s]. Manual verification required.",
+                    targetAddress, line, scannedKey,
+                )            }
+            return nil
+        }
+    }
+
+    file, err := os.OpenFile(knownHostsPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+    if err != nil {
+        return fmt.Errorf("failed to open known_hosts file: %w", err)
+    }
+    defer file.Close()
+
+    if _, err := file.WriteString(scannedKey + "\n"); err != nil {
+        return fmt.Errorf("failed to write to known_hosts file: %w", err)
+    }
+
+    return nil
+}
+
+
+func NewSSHClient(target Target) (*SSHClient, error) {
+    usr, err := user.Current()
+    if err != nil {
+        return nil, fmt.Errorf("unable to determine current user: %w", err)
+    }
+
+    knownHostsPath := filepath.Join(usr.HomeDir, ".ssh", "known_hosts")
+
+    // Ensure known_hosts exists and is valid
+    if err := ensureKnownHosts(knownHostsPath, target.Address); err != nil {
+        return nil, fmt.Errorf("failed to ensure known_hosts: %w", err)
+    }
+
+    hostKeyCallback, err := knownhosts.New(knownHostsPath)
+    if err != nil {
+        return nil, fmt.Errorf("failed to initialize known_hosts verification: %w", err)
+    }
+
+    // Build SSH ClientConfig
+    config := &ssh.ClientConfig{
+        User:            target.Username,
+        Auth:            []ssh.AuthMethod{},
+        HostKeyCallback: hostKeyCallback,
+        Timeout:         time.Duration(target.CommandTimeout) * time.Second,
+    }
+
+    // Add Password Authentication
+    if target.Password != "" {
+        config.Auth = append(config.Auth, ssh.Password(target.Password))
+    }
+
+    // Add Private Key Authentication
+    if target.KeyFile != "" {
+        key, err := os.ReadFile(target.KeyFile)
+        if err != nil {
+            return nil, fmt.Errorf("unable to read private key file %s: %w", target.KeyFile, err)
+        }
+        signer, err := ssh.ParsePrivateKey(key)
+        if err != nil {
+            return nil, fmt.Errorf("unable to parse private key: %w", err)
+        }
+        config.Auth = append(config.Auth, ssh.PublicKeys(signer))
+    }
+
+    // Validate at least one auth method
+    if len(config.Auth) == 0 {
+        return nil, fmt.Errorf("no valid authentication method provided (password or private key)")
+    }
+
+    return &SSHClient{
+        config:  config,
+        host:    target.Address,
+        port:    target.Port,
+        logger:  log.NewNopLogger(),
+        timeout: time.Duration(target.CommandTimeout) * time.Second,
+    }, nil
+}
+
+
+
+func (c *SSHClient) RunCommand(command string) (string, error) {
```

Comment:
Are there any SSH commands that should not be allowed? I realise this may be hard to enforce, but I just think some users might want extra security and I wonder if there are any further checks we can do.